manifests/repo/redhat/redhat.pp

```diff
@@ -99,7 +99,7 @@
 
     $rdo_hash = {
       'rdo-release' => {
-        'baseurl'  => "${centos_mirror_url}/centos/7.4.1708/cloud/\$basearch/openstack-${release}/",
+        'baseurl'  => "http://vault.centos.org/7.4.1708/cloud/x86_64/openstack-newton/",
```

Comment:
- This is the only line where $centos_mirror_url is used in any .pp file so if you remove it, it should be removed from the class parameters.
- Hardcoding "newton" once here and a 2nd time in params.pp is a bit awkward and (imHo) makes the code less maintainable.

All in all I would prefer to revert this whole line back to it's previous state and just replace http://mirror.centos.org with http://vault.centos.org in the class parameters. But can also listen to compelling arguments why this should really be done like this :)

Reply:
* That is not the only line the $centos_mirror_url is used, it is also used on line 120.
* "Hardcoded newton"... ...."makes the code less maintainable". 
* The baseurl also contains an extra 'centos' which mean that I would need to change the baseurl anyway, or make cents_mirror_url less pleasing by appending a '/centos to it. I prefer to change the baseurl.